bundles/org.eclipse.emfcloud.modelserver.common/src/org/eclipse/emfcloud/modelserver/common/patch/AbstractJsonPatchHelper.java

```diff
@@ -187,31 +187,35 @@ protected Command getRemoveCommand(final String modelURI, final ResourceSet reso
       String jsonPath = path.asText();
       SettingValue setting = getSetting(modelURI, resourceSet, jsonPath);
 
+      // If neither the index nor the feature are present, we delete an Object
+      EObject eObject = setting.getEObject();
+
       // If the index is present, we remove a value from a list
       if (setting.getIndex().isPresent()) {
          int index = setting.getIndex().get();
          if (index == CommandParameter.NO_INDEX) {
             // If the index is "-", remove the last value from the list
-            Object currentValue = setting.getEObject().eGet(setting.getFeature());
+            Object currentValue = eObject.eGet(setting.getFeature());
             if (currentValue instanceof Collection) {
                index = ((Collection<?>) currentValue).size() - 1;
             }
          }
-         return RemoveCommand.create(getEditingDomain(setting.getEObject()), setting.getEObject(),
-            setting.getFeature(), index);
+         return RemoveCommand.create(getEditingDomain(eObject), eObject, setting.getFeature(), index);
+      }
+
+      // if we are in a feature which is a list, remove the element from said feature
+      if (eObject.eContainingFeature() != null && eObject.eContainingFeature().isMany()) {
+         return RemoveCommand.create(getEditingDomain(eObject), eObject.eContainer(), eObject.eContainingFeature(),
+            eObject);
       }
 
       // If the feature is present, but index is absent, we unset the value (i.e. set it to default value)
       if (setting.getFeature() != null) {
          Object defaultValue = getDefaultValue(setting.getFeature());
-         return SetCommand.create(getEditingDomain(setting.getEObject()), setting.getEObject(), setting.getFeature(),
-            defaultValue);
+         return SetCommand.create(getEditingDomain(eObject), eObject, setting.getFeature(), defaultValue);
```

Comment:
The canonical way to unset a feature is to initialize a `SetCommand` with the `SetCommand.UNSET_VALUE` token. This ensures that, if the feature is "unsettable" (meaning that it tracks a set/unset state), then it reverts to the unset state. Otherwise it is just being set to a value that happens to be the default.